agent/proxycfg/manager.go

```diff
@@ -158,7 +158,7 @@ func (m *Manager) Register(id ProxyID, ns *structs.NodeService, source ProxySour
 
 func (m *Manager) register(id ProxyID, ns *structs.NodeService, source ProxySource, token string, overwrite bool) error {
 	state, ok := m.proxies[id]
-	if ok {
+	if ok && !state.stoppedRunning() {
```

Comment:
A stopped watcher will be replaced in all situations. Also, if it's already stopped we can skip the `Close()` call as it was already called.